### DISCUSSION TITLE
How to control InputPicker value and menu box

### DISCUSSION THREAD

**Discussion by mirik999** [rsuite/rsuite#2505] [Q&A]
Hi, i have a question about input picker value ( controlled from state ) and menu box. 
1. When i click to input picker i cant correct the value  of input( misspells or something else ) i have to write value again ( caret always at the beginning of input ). PM is killing me 😆 
2. How to open Menu only when i type ( i have figured out with difficult way but if there is some legal or easy way please share )

**Comment by SevenOutman** [comment]
> How to open Menu only when i type ( i have figured out with difficult way but if there is some legal or easy way please share )

Sounds like you're looking for [AutoComplete](https://rsuitejs.com/components/auto-complete/).

**Reply by mirik999** [reply]
No , i am not looking for Autocomplete. I just need to edit value when i click to InputPicker, i have to enter value again when i want to correct one word. Is it possible with InputPicker ? It is very necessary for ux.

**Reply by SevenOutman** [reply]
> i have to enter value again when i want to correct one word. Is it possible with InputPicker ? 

InputPicker is for choosing among several optional values, not for composing random text. How do you edit one word in that case?

**Reply by mirik999** [reply]
If input value did not found in the list then user can keep input value what he wrote.

P.S. But with manipulation DOM i solved the issue for me.
